wgsl/extract-grammar.py

```diff
@@ -510,7 +761,321 @@ def read_spec(options):
                     scanner_i += scanner_parse[-1]  # Advance line index
         scanner_i += 1
 
-    result[scanner_rule.name()]["_comment"] = [["`/\\/\\/.*/`"]]
+    RULE_NAME_CODE_POINTS = string.ascii_lowercase + string.digits + "_"
+    bnfdialect_data = ""
+
+    # Read the contents of the bnfdialect file
+    with open(options.syntax_filename, "r") as file:
+        bnfdialect_data = file.read()
+
+    scan_in = {"name": "blank", "value": "", "content": []}  # blank, comment, rule
+
+    syntax_dict = {}
+
+    s_i = 0
+    s_v = bnfdialect_data[s_i]
+    while s_i < len(bnfdialect_data):
+        if scan_in["name"] == "blank":
+            if s_v == "/" and s_i + 1 < len(bnfdialect_data):  # Comment start
+                if bnfdialect_data[s_i + 1] == "*":
+                    scan_in["name"] = "comment"
+                    scan_in["value"] = 1
+                    scan_in["content"] = []
+                    s_i += 2
+                else:
+                    s_i += 1
+            elif s_v in RULE_NAME_CODE_POINTS:  # Rule start
+                is_rule = False
+                rule_value = s_v
+                s_j = s_i + 1
+                while s_j < len(bnfdialect_data):
+                    if bnfdialect_data[s_j] in RULE_NAME_CODE_POINTS:
+                        rule_value += bnfdialect_data[s_j]
+                        s_j += 1
+                    elif bnfdialect_data[s_j] == ":":
+                        is_rule = True
+                        s_j += 1
+                        break
+                    # Return false when line or rule ends, pattern or token starts
+                    elif bnfdialect_data[s_j] in ["\n", ";", "/", "'"]:
+                        is_rule = False
+                        break
+                    else:
+                        s_j += 1
+                if is_rule:
+                    scan_in["name"] = "rule"
+                    scan_in["value"] = rule_value
+                    scan_in["content"] = [0, 0]
+                    s_i = s_j
+                    syntax_dict[rule_value] = {
+                        "type": "",
+                        "members": [{"type": "sequence", "members": []}],
+                    }
+                else:
+                    s_i += 1
+            else:
+                s_i += 1
```

Comment:
No, these were not errors because the loop does not push anywhere else, moving forward is the responsibility of specific matching case. But since we need to move to line-by-line reading style, I think these code will just go away.